apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -275,6 +281,57 @@ impl Plugin for Telemetry {
             ::tracing::warn!("telemetry.instrumentation.spans.mode is currently set to 'deprecated', either explicitly or via defaulting. Set telemetry.instrumentation.spans.mode explicitly in your router.yaml to 'spec_compliant' for log and span attributes that follow OpenTelemetry semantic conventions. This option will be defaulted to 'spec_compliant' in a future release and eventually removed altogether");
         }
 
+        let graphql_custom_instruments = if cfg!(test) {
```

Comment:
Can we factor out this section of code and the related similar one in activate to a separate `fn` and remove the `cfg!(test)` logic? It doesn't matter if we instruments before activate as no-ops.

Reply:
Fixed https://github.com/apollographql/router/pull/5629/commits/c4a8b14787030df5190bcecc59b4d53fdeb99dfb